stories/form-group/form-group.stories.js

```diff
@@ -36,3 +36,125 @@ export const required = () => `<div class="fd-form-group">
         </div>
     </div>
 `;
+
+export const groupHeader = () => `
+    <div class="fd-form-group">
+        <div class="fd-form-group__header">
+            <span class="fd-form-group__header-text">Group Header</span>
```

Comment:
I see you checked off the a11y requirements on the PR template, what did you test this with? I'm not able to get this read out when accessing the form with VoiceOver. I think it needs to be associated using `aria-labelledby`. See Approach 2 here for an example:

https://www.w3.org/WAI/tutorials/forms/grouping/

Reply:
pressing Ctrl+Option+Shift+Right Arrow read out the header text for me in VoiceOver.

Reply:
added `aria-labelledby` with `id` and `role=group` to get VoiceOver to read the group name on entering the group.